### PR TITLE
TypeError in test_multidut_pfc_pause_lossy_with_snappi.py

### PR DESCRIPTION
**Pull request by sreejithsreekumaran**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

#### What is the motivation for this PR?

Issue #15799 

#### How did you do it?

Made code changes

#### How did you verify/test it?

Verified on a Ixia run

```
------------------------------------------------------------------------------------------------------------------------------------ live log sessionfinish -------------------------------------------------------------------------------------------------------------------------------------
22:33:44 __init__.pytest_terminal_summary         L0067 INFO   | Can not get Allure report URL. Please check logs
==================================================================================================================================== short test summary info ====================================================================================================================================
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info0-aaa14-ixia-m64|0]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info0-aaa14-ixia-m64|1]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info0-aaa14-ixia-m64|2]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info0-aaa14-ixia-m64|5]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info0-aaa14-ixia-m64|6]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info1-aaa14-ixia-m64|0]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info1-aaa14-ixia-m64|1]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info1-aaa14-ixia-m64|2]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info1-aaa14-ixia-m64|5]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info1-aaa14-ixia-m64|6]
========================================================================================================================= 10 passed, 13 warnings in 1932.91s (0:32:12) ==========================================================================================================================
```

### CODE REVIEW

**Comment by sreejithsreekumaran** [comment]
@sdszhang FYI

**Comment by sdszhang** [comment]
@sreejithsreekumaran can you attach the full result for  test_multidut_pfc_pause_lossy_with_snappi.py ?

File: tests/common/snappi_tests/traffic_generation.py [hunk 1]
```diff
@@ -529,8 +530,14 @@ def verify_basic_test_flow(flow_metrics,
         else:
             pytest_assert(tx_frames == rx_frames,
                           "{} should not have any dropped packet".format(metric.name))
-
-            exp_test_flow_rx_pkts = data_flow_config["flow_rate_percent"] / 100.0 * speed_gbps \
+            # Check if flow_rate_percent is a dictionary
+            if isinstance(data_flow_config["flow_rate_percent"], dict):
+                # Sum all the flow rate percentages
+                total_flow_rate_percent = sum(data_flow_config["flow_rate_percent"].values())
```
Comment (sdszhang) [inline tests/common/snappi_tests/traffic_generation.py:536]: if there are multiple test flow, should we use individual flow rate instead of summary?

Comment (sreejithsreekumaran) [reply tests/common/snappi_tests/traffic_generation.py]: done

**Comment by sreejithsreekumaran** [comment]
> @sreejithsreekumaran can you attach the full result for test_multidut_pfc_pause_lossy_with_snappi.py ?

```
------------------------------------------------------------------------------------------------------------------------------------ live log sessionfinish -------------------------------------------------------------------------------------------------------------------------------------
12:23:43 __init__.pytest_terminal_summary         L0067 INFO   | Can not get Allure report URL. Please check logs
==================================================================================================================================== short test summary info ====================================================================================================================================
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info0-aaa14-ixia-m64|0]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info0-aaa14-ixia-m64|1]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info0-aaa14-ixia-m64|2]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info0-aaa14-ixia-m64|5]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info0-aaa14-ixia-m64|6]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info1-aaa14-ixia-m64|0]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info1-aaa14-ixia-m64|1]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info1-aaa14-ixia-m64|2]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info1-aaa14-ixia-m64|5]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio[multidut_port_info1-aaa14-ixia-m64|6]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_multi_lossy_prio[multidut_port_info0]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_multi_lossy_prio[multidut_port_info1]
SKIPPED [2] snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py:133: Reboot type warm is not supported on cisco-8000 switches
SKIPPED [2] snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py:133: Reboot type fast is not supported on cisco-8000 switches
SKIPPED [2] snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py:193: Reboot type warm is not supported on cisco-8000 switches
SKIPPED [2] snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py:193: Reboot type fast is not supported on cisco-8000 switches
ERROR snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio_reboot[multidut_port_info0-cold-aaa14-ixia-m64|1] - Failed: Processes "['save_config_and_reboot--<MultiAsicSonicHost aaa14-ixia-m64>']" failed with exit code ""
ERROR snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_single_lossy_prio_reboot[multidut_port_info1-cold-aaa14-ixia-m64|1] - Failed: Processes "['save_config_and_reboot--<MultiAsicSonicHost aaa14-ixia-m64>']" failed with exit code ""
ERROR snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_multi_lossy_prio_reboot[multidut_port_info0-cold] - Failed: Processes "['save_config_and_reboot--<MultiAsicSonicHost aaa14-ixia-m64>']" failed with exit code ""
ERROR snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossy_with_snappi.py::test_pfc_pause_multi_lossy_prio_reboot[multidut_port_info1-cold] - Failed: Processes "['save_config_and_reboot--<MultiAsicSonicHost aaa14-ixia-m64>']" failed with exit code ""
============================================================================================================== 12 passed, 8 skipped, 16 warnings, 4 errors in 13049.32s (3:37:29) ===============================================================================================================
```

**Review by sdszhang — APPROVED** [review]

**Pull request merged**